src/praisonai-train/praisonai_train/train/llm/trainer.py

```
@@ -556,6 +657,15 @@ def process_dataset(self, dataset_info):
     def load_datasets(self):
         datasets = []
         for dataset_info in self.config["dataset"]:
+            # Advertised in the shipped templates but never read. Saying so beats
+            # silently discarding a formatter the user believed was running.
+            if dataset_info.get("processing_func"):
+                logger.warning(
```

Ruff (0.16.2)

[error] Undefined name `logger`

(F821)